elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideDynamicConfiguration.java

```diff
@@ -0,0 +1,164 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.persistence.Entity;
+import javax.persistence.spi.PersistenceUnitInfo;
+import javax.sql.DataSource;
+
+/**
+ * Dynamic Configuration For Elide Services. Override any of the beans (by
+ * defining your own) and setting flags to disable in properties to change the
+ * default behavior.
+ */
+
+@Slf4j
+@Configuration
+@EnableConfigurationProperties(ElideConfigProperties.class)
+@ConditionalOnExpression("${elide.dynamic-config.enabled:false}")
+public class ElideDynamicConfiguration {
+
+    public static final String HIBERNATE_DDL_AUTO = "hibernate.hbm2ddl.auto";
+    public static final String HIBERNATE_PHYSICAL_NAMING = "hibernate.physical_naming_strategy";
+    public static final String HIBERNATE_IMPLICIT_NAMING = "hibernate.implicit_naming_strategy";
+    public static final String HIBERNATE_ID_GEN_MAPPING = "hibernate.use-new-id-generator-mappings";
+
+    /**
+     * Configure factory bean to create EntityManagerFactory for Dynamic Configuration.
+     * @param source :DataSource for JPA
+     * @param jpaProperties : JPA Config Properties
+     * @param hibernateProperties : Hibernate Config Properties
+     * @param dynamicCompiler : ElideDynamicEntityCompiler
+     * @return LocalContainerEntityManagerFactoryBean bean
+     */
+    @Bean
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory (
+            DataSource source,
+            JpaProperties jpaProperties,
+            HibernateProperties hibernateProperties,
+            ObjectProvider<ElideDynamicEntityCompiler> dynamicCompiler) {
+
+            //Map for Persistent Unit properties
+            Map<String, Object> puiPropertyMap = new HashMap<>();
+
+            //Bind entity classes from classpath to Persistence Unit
+            ArrayList<Class> bindClasses = new ArrayList<>();
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(Entity.class));
+
+            //Bind FromTable/FromSubSelect classes from classpath to Persistence Unit
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(FromTable.class));
+            bindClasses.addAll(ClassScanner.getAnnotatedClasses(FromSubquery.class));
+
+            //Map of JPA Properties to be be passed to EntityManager
+            Map<String, String> jpaPropMap = jpaProperties.getProperties();
+
+            String hibernateGetDDLAuto = hibernateProperties.getDdlAuto();
+            Naming hibernateGetNaming =  hibernateProperties.getNaming();
+            String hibernateImplicitStrategy = hibernateGetNaming.getImplicitStrategy();
+            String hibernatePhysicalStrategy = hibernateGetNaming.getPhysicalStrategy();
+            Boolean hibernateGetIdenGen = hibernateProperties.isUseNewIdGeneratorMappings();
+
+            //Set the relevant property in JPA corresponding to Hibernate Property Value
+            hibernateJPAPropertyOverride(jpaPropMap, HIBERNATE_DDL_AUTO, hibernateGetDDLAuto);
```

Comment:
Why can't we copy all of the properties from hibernateProperties instead of setting explicit ones.  THere are dozens of hibernate properties that might be set.

Reply:
Discussed with Aaron, this can be ignored.